ops/model.py

```diff
@@ -119,6 +122,54 @@
         'egress-subnets': List[str]
     })
 
+    _SecretMetadataDict = TypedDict('_SecretMetadataDict', {
+        'label': str,
+        'revision': str,
+        'expires': str,
+        'rotation': str,
+
+        # internally converted to SecretRotationPolicy enum
+        'rotates': Literal['never', 'hourly', 'daily', 'weekly',
+                           'monthly', 'quarterly', 'yearly'],
+    })
+
+    _SecretGetMetadataDict = Dict[str, _SecretMetadataDict]
+    _SecretGetContentsDict = Dict[str, Dict[str, str]]
+    _SecretGetResponseType = Union[str,
+                                   Dict[str, Union[
+                                       '_SecretGetContentsDict',
+                                       '_SecretGetMetadataDict']]]
+
+
+class SecretRotate(enum.Enum):
+    """Allowed secret rotation policies."""
+    NEVER = 'never'  # the default in juju
+    HOURLY = 'hourly'
+    DAILY = 'daily'
+    WEEKLY = 'weekly'
+    MONTHLY = 'monthly'
+    QUARTERLY = 'quarterly'
+    YEARLY = 'yearly'
+
+    @staticmethod
+    def cast(value: Optional[Union['SecretRotate', str]]):
+        """Cast :param:`value` to :class:`SecretRotate`.
+
+        raise SecretRotateValueError on failure.
+        """
+        if value is None:
+            return None  # None will be defaulted to 'never' on the juju side.
+        try:
+            return SecretRotate(value)
+        except ValueError as e:
+            raise SecretRotateValueError(value) from e
+
+
+class SecretOwner(enum.Enum):
+    """Secret owner names."""
```

Comment:
We should expand this comment - application owned secrets require the leader to manage them etc.